### PR TITLE
End to end tests

### PR DESCRIPTION
**Pull request by bertvi**
A minor fix to subprocess usage related to api keys, namely substituting subprocess.run with subprocess.getoutput to make it compatible with Python 3.6+. In addition, "root" key has been changed to "niis" where SSH access to security server is performed and "reset_security_servers" script has been amended to also take ssh user as parameter, currently "niis" should be given ti make it compatible with the source code, but in an upcoming user story the SSH user will be made configurable from the YAML.

### CODE REVIEW

File: reset_security_server.sh [hunk 1]
```diff
@@ -64,12 +65,18 @@ add_public_key() {
     for name in $names
     do
         printf "\nAdding public key to LXD container %s\n" "$name"
-        lxc exec "$name" -- bash -c "mkdir -p $3"
-        lxc exec "$name" -- bash -c "echo \"$(cat "$2")\" > $3/$4"
+        lxc exec "$name" -- bash -c "useradd -c \"$2 user\" -d /home/$2 -s /bin/bash $2"
+        lxc exec "$name" -- bash -c "echo \"$2 ALL=(ALL) NOPASSWD: ALL\" > /etc/sudoers.d/$2"
+        lxc exec "$name" -- bash -c "mkdir -p /home/$2/$4"
+        lxc exec "$name" -- bash -c "echo \"$(cat "$3")\" > /home/$2/$4/$5"
```
Comment (unserializable) [inline reset_security_server.sh]: Replace $N with local variables here and below, ``"echo \"$(cat "$3")\" > /home/$2/$4/$5"``is opaque.

File: reset_security_server.sh [hunk 2]
```diff
@@ -64,12 +65,18 @@ add_public_key() {
     for name in $names
     do
         printf "\nAdding public key to LXD container %s\n" "$name"
-        lxc exec "$name" -- bash -c "mkdir -p $3"
-        lxc exec "$name" -- bash -c "echo \"$(cat "$2")\" > $3/$4"
+        lxc exec "$name" -- bash -c "useradd -c \"$2 user\" -d /home/$2 -s /bin/bash $2"
+        lxc exec "$name" -- bash -c "echo \"$2 ALL=(ALL) NOPASSWD: ALL\" > /etc/sudoers.d/$2"
+        lxc exec "$name" -- bash -c "mkdir -p /home/$2/$4"
```
Comment (unserializable) [inline reset_security_server.sh]: Could be already created with ``useradd -m`` for all Debian based systems. If not sufficient for Redhat, then ignore the note.

File: tests/end_to_end/tests.py [hunk 3]
```diff
@@ -140,8 +140,8 @@ def revoke_api_key(self):
         curl_cmd = "curl -X DELETE -u " + self.config["api_key"][0]["credentials"] + " --silent " + \
                    self.config["api_key"][0]["url"] + "/" + str(base.api_key_id[self.config["security_server"][0]['name']]) + " -k"
         cmd = "ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i \"" + \
-              self.config["api_key"][0]["key"] + "\" root@" + self.config["security_server"][0]["name"] + " \"" + curl_cmd + "\""
-        subprocess.run(cmd, shell=True, check=False, capture_output=True)
+              self.config["api_key"][0]["key"] + "\" niis@" + self.config["security_server"][0]["name"] + " \"" + curl_cmd + "\""
+        subprocess.getoutput(cmd)
```
Comment (unserializable) [inline tests/end_to_end/tests.py]: Use ``getstatusoutput``, fail with with error message if unsuccessful. Replace ``Exception`` catch with narrowed one, either here or in SSTOOLKIT-56.

**Review by unserializable — CHANGES_REQUESTED** [review]

**Comment by unserializable** [comment]
After succesful run of ``reset_security_server.sh`` access is  succesful, key revocation fails, most likely due to now unmatching user accounts:

```
$ xrdsst  -c ss8.yaml apply
API key "65de40f7-15a6-4aa8-9dec-aaccededb9f8" for security server ss8 created.
AUTO ['init']->'ss8'
... SNIP ...
  File "/home/user/dev/niis/clean36/X-Road-Security-Server-toolkit/xrdsst/main.py", line 194, in revoke_api_key
    process = subprocess.run(cmd, shell=True, check=True, capture_output=True)
  File "/usr/lib/python3.8/subprocess.py", line 512, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "/home/user/.ssh/id_ed25519" root@ss8 "curl -X DELETE -u xrd:secret --silent https://localhost:4000/api/v1/api-keys/1 -k"' returned non-zero exit status 255.
````

**Comment by bertvi** [comment]
> After succesful run of `reset_security_server.sh` access is succesful, key revocation fails, most likely due to now unmatching user accounts:
> 
> ```
> $ xrdsst  -c ss8.yaml apply
> API key "65de40f7-15a6-4aa8-9dec-aaccededb9f8" for security server ss8 created.
> AUTO ['init']->'ss8'
> ... SNIP ...
>   File "/home/user/dev/niis/clean36/X-Road-Security-Server-toolkit/xrdsst/main.py", line 194, in revoke_api_key
>     process = subprocess.run(cmd, shell=True, check=True, capture_output=True)
>   File "/usr/lib/python3.8/subprocess.py", line 512, in run
>     raise CalledProcessError(retcode, process.args,
> subprocess.CalledProcessError: Command 'ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o LogLevel=ERROR -i "/home/user/.ssh/id_ed25519" root@ss8 "curl -X DELETE -u xrd:secret --silent https://localhost:4000/api/v1/api-keys/1 -k"' returned non-zero exit status 255.
> ```

Should be fixed now.

**Pull request merged**